lnrpc/lightning.proto

```diff
@@ -2163,13 +2163,36 @@ message CloseChannelRequest {
     // NOTE: This field is only respected if we're the initiator of the channel.
     uint64 max_fee_per_vbyte = 7;
 
-    // If true, then the rpc call will not block while it awaits a closing txid
-    // to be broadcasted to the mempool. To obtain the closing tx one has to
-    // listen to the stream for the particular updates. Moreover if a coop close
-    // is specified and this flag is set to true the coop closing flow will be
-    // initiated even if HTLCs are active on the channel. The channel will wait
-    // until all HTLCs are resolved and then start the coop closing process. The
-    // channel will be disabled in the meantime and will disallow any new HTLCs.
+    // If `no_wait=true` (`wait=false`) when a coop close is attempted, then the
+    // rpc call will not initially block while it awaits a closing txid to be
+    // broadcasted to the mempool. Instead, a coop close will be initiated even
+    // if there are HTLCs active in flight on the channel and a `close_instant`
+    // will be immediately returned which contains `num_pending_htlcs` if there
+    // are in flight HTLCs that need to be taken out of flight before a coop
+    // close can be initiated. After the `close_instant` is received, then the
+    // channel will be disabled to disallow any new HTLCs and the rpc will block
+    // while it waits for the exsting in flight HTLCs to be resolved and then
+    // start the coop closing process. To obtain the closing TXID one has to
+    // listen to the stream for a `close_pending` update. If the client
+    // disconnects the stream before this is received, the rpc will continue to
+    // wait for the in flight HTLCs to be resolved and then start the coop
+    // closing process. Note: `lncli closechannel` always sets `no_wait=true`
+    // and its `--block` option controls if `lncli` should wait for a
+    // `close_pending` update and a `chan_close` update before returning or if
+    // it should disconnect the stream and return immediately after
+    // `close_instant` is received. If `no_wait=false` (`wait=true`) when a coop
+    // close is attempted and there are in flight HTLCs on the channel, then the
+    // rpc call will fail and abort the closing process because it can't wait
+    // for a closing TXID to be broadcast to the mempool because it needs to
+    // immediately tell the user that there are in flight HTLCs that need to be
+    // resolved before that can happen. If a coop close is attempted with no in
+    // flight HTLCs on the channel OR a force close is attempted, the `no_wait`
+    // option is ignored. In summary, the `no_wait` option controls if the user
```

Comment:
If a coop close is attempted with no inflight HTLCs on the channel the `no_wait=true` parameter has still the effect that the stream sends the `close_instant` msg immediately however that should not make much of a difference because LND should broadcast the closing tx without much time delay and therefore should send the `close_pending ` update right after.